### PR TITLE
Check HTTPS health and enforcement

### PR DESCRIPTION
**Pull request by benbalter**
Fixes https://github.com/github/pages-health-check/issues/62.

This adds three new methods to the `domain` object:

* `https?` - does this domain respond over HTTPS with a valid cert?
* `enforces_https` - does this domain redirect from HTTP to HTTPS?
* `https_error` - the HTTPS cert error, if any

### CODE REVIEW

File: spec/github_pages_health_check_domain_spec.rb [hunk 1]
```diff
@@ -474,4 +474,89 @@ def mx_packet
...
+    let(:domain) { "pages.github.com" }
+    let(:domain_check) { make_domain_check(domain) }
+
+    context "a site that supports HTTPS" do
+      let(:domain) { "pages.github.com" }
+
+      before do
+        stub_request(:head, "https://pages.github.com/").
+           to_return(:status => 200, :headers => {:server => "GitHub.com"})
+         allow(domain_check.send(:https_response)).to receive(:return_code) { :ok }
```
Comment (parkr) [inline spec/github_pages_health_check_domain_spec.rb]: Why is this needed if you stub the request?

File: spec/github_pages_health_check_domain_spec.rb [hunk 2]
```diff
@@ -474,4 +474,89 @@ def mx_packet
...
+      end
+    end
+
+    context "a site that enforces HTTPS" do
+      let(:domain) { "pages.github.com" }
+
+      before do
+        stub_request(:head, "https://pages.github.com/").
+           to_return(:status => 200, :headers => {:server => "GitHub.com"})
+         allow(domain_check.send(:https_response)).to receive(:return_code) { :ok }
```
Comment (parkr) [inline spec/github_pages_health_check_domain_spec.rb]: Why do we need this?

Comment (parkr) [inline spec/github_pages_health_check_domain_spec.rb]: Why is this needed?

File: spec/github_pages_health_check_domain_spec.rb [hunk 3]
```diff
@@ -474,4 +474,89 @@ def mx_packet
...
+      end
+    end
+
+    context "a site that doesn't support HTTPS" do
+      let(:domain) { "pages.github.com" }
+
+      before do
+        stub_request(:head, "https://pages.github.com/").
+           to_return(:status => 200, :headers => {:server => "GitHub.com"})
+         allow(domain_check.send(:https_response)).to receive(:return_code) { :ssl_cacert }
```
Comment (parkr) [inline spec/github_pages_health_check_domain_spec.rb]: Why is this needed?

File: spec/github_pages_health_check_domain_spec.rb (hunk 2 above)
Comment (parkr) [inline spec/github_pages_health_check_domain_spec.rb]: Why is this needed?

Comment (parkr) [inline spec/github_pages_health_check_domain_spec.rb]: Why is this needed?

File: spec/github_pages_health_check_domain_spec.rb [hunk 4]
```diff
@@ -474,4 +474,89 @@ def mx_packet
...
+      end
+    end
+
+    context "a site with a relative redirect" do
+      let(:domain) { "pages.github.com" }
+
+      before do
+        stub_request(:head, "https://pages.github.com/").
+           to_return(:status => 200, :headers => {:server => "GitHub.com"})
+         allow(domain_check.send(:https_response)).to receive(:return_code) { :ok }
```
Comment (parkr) [inline spec/github_pages_health_check_domain_spec.rb]: Why is this needed?

File: spec/github_pages_health_check_domain_spec.rb (hunk 1 above)
Comment (benbalter) [inline spec/github_pages_health_check_domain_spec.rb]: We're using Webmock the way it's describes, and `return_code` is where the SSL errors are exposed, but looking at the raw object, it seems Webmock doesn't stub the `response_code`.
